src/middleware/apiKeyChecker.ts

```diff
@@ -0,0 +1,29 @@
+import fs from 'fs';
+import { AuthenticationError } from '../errors';
+import type {
+  NextFunction,
+  Request,
+  Response,
+} from 'express';
+
+export const checkApiKey = (req: Request, res: Response, next: NextFunction): void => {
+  const apiKey = req.headers['x-api-key'] ?? '';
+  if (apiKey === '') {
+    next(new AuthenticationError(
+      'API key not provided in the header "x-api-key"',
+    ));
+  } else {
+    try {
+      const data = fs.readFileSync('test_keys.txt', 'utf8').split('\n');
```

Comment:
OK, now I see why it was not originally `test_keys.txt` :smile:

Hmm, I'm still thinking about how to resolve this. What went wrong with the tests when using an env var?

Reply:
FWIW we are able to use an environment variables in tests using the type of mocking technique described in https://www.webtips.dev/how-to-mock-processenv-in-jest

Reply:
I'm trying it out locally to see what you mean.

Reply:
Good call -- and yes agreed they should supply the key